agent/engine/execcmd/manager.go

```diff
@@ -78,11 +79,18 @@ type manager struct {
 }
 
 func NewManager() *manager {
-	return &manager{
+	m := &manager{
 		hostBinDir:          HostBinDir,
 		retryMaxDelay:       defaultRetryMaxDelay,
 		retryMinDelay:       defaultRetryMinDelay,
 		startRetryTimeout:   defaultStartRetryTimeout,
 		inspectRetryTimeout: defaultInspectRetryTimeout,
 	}
+	return m
+}
+
+func NewManagerWithBinDir(hostBinDir string) *manager {
```

Comment:
nit: maybe this could just be an overload of `NewManager(hostBinDir string)`, might not be necessary to have a new name

Reply:
There is no method overloading in golang =)

Reply:
thanks for the explanation!!